xapian-core/api/omenquire.cc

```diff
@@ -452,6 +464,37 @@ MSet::Internal::read_docs() const
     requested_docs.clear();
 }
 
+void
+MSet::Internal::set_item_weight(Xapian::doccount i, double wt_)
+{
+    /* max_attained is updated assuming that set_item_weight is called on every Xapian::doccount from 0 to mset.size() in an ascending order.
+     * While assigning new weights max_attained is updated as the maximum of the new weights set till Xapian::doccount i.
+     */
+    if (i == 0)
+	max_attained = wt_;
+    else
+	max_attained = max(max_attained, wt_);
+    /* While assigning new weights max_possible is updated as the maximum of max_attained at Xapian::doccount i
+     * and max_possible at Xapian::doccount i-1. While calculating max_possible for Xapian::doccount 0 the max_possible
+     * of the previous weights is taken into account.
+     * max_possible is set as the maximum of the new weights as well as the old weights.
+     * Ideally the max_possible should be the maximum possible weight that can be assigned by the resorting algorithm
+     * but since it is not possible to calculate max possible weight for different algorithms we use this approach.
```

Comment:
:+1: .... I have updated the docstring and the comments ..... I will use them as they are if you approve